cmake/SetupProtobuf.cmake

```diff
@@ -24,8 +24,8 @@ if(LINK_STATIC_LIBPROTOBUF)
             URL_HASH ${PROTOBUF_HASH}
             DOWNLOAD_DIR "${DOWNLOAD_LOCATION}"
             BUILD_IN_SOURCE 1
-            CONFIGURE_COMMAND ./configure --prefix=${PROTOBUF_DIR} CFLAGS=${PROTOBUF_CFLAGS} CXXFLAGS=${PROTOBUF_CXXFLAGS}
-            BUILD_COMMAND make
+            CONFIGURE_COMMAND bash -ex ${CMAKE_MODULE_PATH}/configure-helper.sh ${CMAKE_C_COMPILER} ${CMAKE_CXX_COMPILER} ${PROTOBUF_DIR} ${PROTOBUF_CFLAGS} ${PROTOBUF_CXXFLAGS}
+            BUILD_COMMAND make -j4
```

Comment:
`-j4` option seems not always suitable. How to check the number of available processors here dynamically?
For example: http://d.hatena.ne.jp/embedded/20140829/p1

Reply:
Make sense. I also did `make -j$(proc)` before but it accidentally launches *dozens of* build process (even if `nproc` is `4` in my laptop) and runs out my CPU resources :sweat_smile:. Unfortunately I don't know the clear cause of this behaviour. BTW, I think `-j4` is reasonable choice in many cases because almost modern CPUs have more than two or four cores.

Reply:
I see. So `-j4` is also no problem.

Reply:
Thanks!